--- kind-0.8.1/pkg/cluster/internal/providers/docker/provision.go	2020-05-02 02:54:15.000000000 +0300
+++ kind-0.8.1/pkg/cluster/internal/providers/docker/provision-patched.go	2021-07-06 12:36:23.376743088 +0300
@@ -227,6 +227,8 @@
 		"--volume", "/var",
 		// some k8s things want to read /lib/modules
 		"--volume", "/lib/modules:/lib/modules:ro",
+		// ipc host
+		"--ipc", "host",
 	},
 		args...,
 	)
--- kind-0.8.1/pkg/cmd/kind/version/version.go	2020-05-02 02:54:15.000000000 +0300
+++ kind-0.8.1/pkg/cmd/kind/version/version-patched.go	2021-07-06 12:38:29.521715917 +0300
@@ -50,7 +50,7 @@
 }
 
 // VersionCore is the core portion of the kind CLI version per Semantic Versioning 2.0.0
-const VersionCore = "0.8.1"
+const VersionCore = "0.8.1-patched"
 
 // VersionPreRelease is the pre-release portion of the kind CLI version per
 // Semantic Versioning 2.0.0
