--- kind-0.11.1/pkg/cluster/internal/providers/docker/provision.go	2021-07-02 15:32:23.872508147 +0300
+++ kind-0.11.1-patched/pkg/cluster/internal/providers/docker/provision.go	2021-06-30 18:57:28.806000000 +0300
@@ -243,6 +243,8 @@
 		"--volume", "/var",
 		// some k8s things want to read /lib/modules
 		"--volume", "/lib/modules:/lib/modules:ro",
+		// ipc host
+		"--ipc", "host",
 	},
 		args...,
 	)
--- kind-0.11.1/pkg/cmd/kind/version/version.go	2021-05-28 02:18:13.000000000 +0300
+++ kind-0.11.1-patched/pkg/cmd/kind/version/version.go	2021-07-02 15:31:45.440367546 +0300
@@ -50,7 +50,7 @@
 }
 
 // VersionCore is the core portion of the kind CLI version per Semantic Versioning 2.0.0
-const VersionCore = "0.11.1"
+const VersionCore = "0.11.1-patched"
 
 // VersionPreRelease is the pre-release portion of the kind CLI version per
 // Semantic Versioning 2.0.0
